src/Site.js

```diff
@@ -667,11 +667,15 @@ Site.prototype.regenerateAffectedPages = function (filePaths) {
 /**
  * Uses heading data in built pages to generate heading and keyword information for siteData
  * Subsequently writes to siteData.json
+ * @param filePaths array of paths corresponding to files that have changed
```

Comment:
Using the word "changed" here sound a bit strange (I know it is following the comments in other method, but those methods are usually doing purely regeneration, rather than initial generation + regeneration).

Perhaps the name of the method can be `updateSiteData()` rather than `generateSiteData()`? To make it clearer that the changes are accumulative.